packages/core/src/browser/runOnReadyState.ts

```diff
@@ -14,3 +14,17 @@ export function runOnReadyState(
   const eventName = expectedReadyState === 'complete' ? DOM_EVENT.LOAD : DOM_EVENT.DOM_CONTENT_LOADED
   return addEventListener(configuration, window, eventName, callback, { once: true })
 }
+
+export function asyncRunOnReadyState(
```

Comment:
For now, I introduced an async version of `runOnReadyState` but if we agree on this lazy loading approach I'll refactor the original `runOnReadyState` to use promises.

Reply:
Refactoring `runOnReadyState` is quite complexe. Let's keep `asyncRunOnReadyState` for now